ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp

```diff
@@ -54,7 +54,20 @@ std::vector<ttnn::Tensor> _atan2_bw(
     return grad_tensor;
 }
 
-std::vector<std::optional<ttnn::Tensor>> _addalpha_bw(
+// to be used for all binary backward ops to create a zeros tensor when there's no preallocated output_tensor
+void preallocated_tensors_check(std::optional<Tensor>& input_grad, std::optional<Tensor>& other_grad, const Tensor& input, const Tensor& other,  const std::array<bool, 2>& required_outputs){
+
+    TT_FATAL(required_outputs[0] || required_outputs[1], "required_outputs cannot be all false");
+
```

Comment:
Can we change this to something like...
"At least one gradient is expected to to be calculated."